### ISSUE TITLE
Double Ajax request when filter applied with saveSort / pager / stickyHeaders

### ISSUE THREAD

**Issue by eukal-dh** [Mottie/tablesorter#712]
Hi,

(sorry for my english as it's not my mother tongue but I'ill do my best).

I am using tablesorter with the pager plugin (with Ajax request) and : 
<pre>widgets : ['zebra','saveSort','stickyHeaders','filter']</pre>

(last versions : 2.7.17)

<pre>
src="/js/tablesorter/2.7.17/jquery.tablesorter.min.js
src="/js/tablesorter/2.7.17/jquery.tablesorter.widgets.min.js
src="/js/tablesorter/2.7.17/addons/pager/jquery.tablesorter.pager.min.js
</pre>

My major goal is to save the sort, the page and the filter to enable the user to recover the table in the exact statement he left it. That's why i am using saveSort & savePages

Everything is working well except when I try to recover a table I left a few moment before with an applied filter (data-column="all"). In that case, I get two Ajax request with an uncorrect start page for the second request.

Let me give an example. I filtered "Paris", I was on page number 5, sized 50 results and sorted descending on column C. I went somewhere else in my application and I am coming back to my table. 

In that case, I get two Ajax request. The first one is correct but disappear and is replaced by the second one which has an uncorrect start page : 

<pre>
?page=4&size=50&column[2]=0&filter[13]=paris
?page=0&size=50&column[2]=0&filter[13]=paris
</pre>

Here is my script : 
```js
	$(document).ready(function() {
	    	    
		var $table = $('#demandes-renseignements');
		
		var pagerOptions = {
		    
			// target the pager markup
			container: $(".pager"),
			
			ajaxUrl : 'ajax-demandes-renseignements.php?page={page}&size={size}&{sortList:column}&{filterList:filter}',

			// add more ajax settings here
			// see http://api.jquery.com/jQuery.ajax/#jQuery-ajax-settings
			ajaxObject: {
				dataType: 'json'
			},				

			ajaxProcessing: function(data, table, xhr){

			    if (data && data.hasOwnProperty('rows')) {
				var r, row, c, d = data.rows,
				// total number of rows (required)
				total = data.total_rows,
				// array of header names (optional)
				headers = data.headers,
				// all rows: array of arrays; each internal array has the table cell data for that row
				rows = '',
				// len should match pager set size (c.size)
				len = d.length;
				// this will depend on how the json is set up - see City0.json
				// rows
				for ( r=0; r < len; r++ ) {

				    WEB_N_DEMANDE = d[r]['WEB_N_DEMANDE'];
				    CLASS_TR = d[r]['CLASS_TR'];

				    var invisible   = ["CLASS_TR", "WEB_N_DEMANDE"];
				    var center	    = ["Dossier","Demande","Envoi en mairie","NB_DOC","DPU","Facturation","Paiement"];
				    var align	    = "";

				    rows += '<tr class="' + CLASS_TR + '">';
				    // cells
				    for ( c in d[r] ) {
					if (typeof(c) === "string") {
					    if(c == "DELETE") {
						rows += '<td>' + d[r][c] + '</td>'; // add each table cell data to row
					    } else if(! inArray(c,invisible)) {

						if(inArray(c,center)) { 
						    align = "align=center"; 
						} else { 
						    align = ""; 
						}
						rows += '<td ' + align + ' onclick="javascript:voir_dossier(' + WEB_N_DEMANDE + ');">' + d[r][c] + '</td>'; // add each table cell data to row
					    }
					}
				    }
				    rows += '</tr>';
				}
				// don't attach the $(rows) because it's difficult to tell old from new data
				// and no need to trigger an update method, it's done internally
				return [ total, $(rows), headers ];
			    }
			},

			// output string - default is '{page}/{totalPages}'; possible variables: {page}, {totalPages}, {startRow}, {endRow} and {totalRows}
			output: '{startRow} &agrave; {endRow} / {totalRows} <strong>[ Page : {page} / {totalPages} ]</strong>',

			// apply disabled classname to the pager arrows when the rows at either extreme is visible - default is true
			updateArrows: false,

			// starting page of the pager (zero based index)
			page: 0,

			// Number of visible rows - default is 10
			size: 25,

			// Save pager page & size if the storage script is loaded (requires $.tablesorter.storage in jquery.tablesorter.widgets.js)
			savePages : true,

			// defines custom storage key
			storageKey : 'tablesorter-pager-demandes-renseignements',

			// if true, the table will remain the same height no matter how many records are displayed. The space is made up by an empty
			// table row set to a height to compensate; default is false
			fixedHeight: false,

			// remove rows from the table to speed up the sort of large tables.
			// setting this to false, only hides the non-visible rows; needed if you plan to add/remove rows with the pager enabled.
			removeRows: false,

			// css class names of pager arrows
			cssNext        : '.next',  // next page arrow
			cssPrev        : '.prev',  // previous page arrow
			cssFirst       : '.first', // go to first page arrow
			cssLast        : '.last',  // go to last page arrow
			cssPageDisplay : '.pagedisplay', // location of where the "output" is displayed
			cssPageSize    : '.pagesize', // page size selector - select dropdown that sets the "size" option

			// class added to arrows when at the extremes (i.e. prev/first arrows are "disabled" when on the first page)
			cssDisabled    : 'disabled', // Note there is no period "." in front of this class name
			cssGoto	       : '.gotoPage'
		}
		
		// initialize the pager plugin
		// ****************************
		$table.on('filterInit', function(){
		    $table.tablesorterPager(pagerOptions);
		});
		// Initialize tablesorter
		// ***********************
		$table.tablesorter({
			widgets : ['zebra','saveSort','stickyHeaders','filter'],
			dateFormat: 'ddmmyyyy',
			sortLocaleCompare: true, // needed for accented characters in the data
			selectorHeaders: 'thead th',
			widthFixed:true,
			widgetOptions : {
			    filter_columnFilters	: false,
			    filter_saveFilters		: true,
			    filter_serversideFiltering	: true,
			    filter_external		: '.search'
			}
		});

	});
```

What do I do wrong ?

**Comment by Mottie** [comment]
@eukal-dh!

I apologize again for not following up promptly; I keep forgetting about this issue :disappointed: 

 I'll make a special note to look into this issue for the next update.

**Comment by Mottie** [comment]
Hi @eukal-dh!

I am so sorry for this very long delay in checking on this issue. I just tested this problem with the latest pager plugin and it does not appear to be an issue any longer. I am only seeing one ajax request.

I did notice that when I change the filter value, the page reset to the first, but that is because the [`pageReset`](http://mottie.github.io/tablesorter/docs/#pager-page-reset) is set to `0` (zero). Change it to `false` to keep the page the same, if needed.
